### DISCUSSION TITLE
Is it possible to disable `--verifytypes` for certain symbols?

### DISCUSSION THREAD

**Discussion by RobertCraigie** [microsoft/pyright#6199] [Q&A]
We're in a very niche situation where we explicitly do not want to explicitly provide types as it leads to an insane amount of code duplication for method signatures.

In essence our case is:
```py
class Completions:
  with_raw_response: RawCompletions

  def create(self, foo: str) -> Completion:
    ...

class RawCompletions:
  def __init__(self, completions: Completions) -> None:
    self.create = make_response_wrapper(completions.create)

P = ParamSpec('P')
R = TypeVar('R')

def make_response_wrapper(fn: Callable[P, R]) -> Callable[P, APIResponse[R]]:
  ...
```

In this example duplicating the signature would be totally fine but our method signatures can get pretty large and some methods require overloads which then increases the duplication even further.

We've verified that Pyright, Mypy and PyCharm all infer the types correctly so we think the tradeoff of inferring types vs code duplication is worth it. However, rightly so `--verifytypes` complains about the missing type annotations. I find `--verifytypes` to be a very useful tool so I'd really want to avoid not running it at all.

Is there a way to ignore certain symbols explicitly? e.g.
```py
self.create = make_response_wrapper(completions.create)  # pyright: ignore[verifytypes]
```

**Comment by erictraut** [comment]
Type verification is implemented by walking symbol tables (as opposed to type evaluation, which is done on parse trees). That means `# pyright: ignore` comments cannot be used to control which errors are output from the type verifier.

As a workaround, you could use the `--outputjson` flag to output JSON output rather than text. You can then write a simple script to filter the JSON output however you'd like (e.g. ignore errors for certain symbols on an exclude list).
